_docs/for-contributors/2017-01-01-building-from-source.md

```diff
@@ -39,54 +46,45 @@ Switch to the latest release branch, for example, `cardano-sl-1.0`:
 
     $ git checkout cardano-sl-1.0
 
-Then enter `nix-shell`:
-
-    $ nix-shell
-
-And if it is the first project in Haskell on this machine, run `stack setup`:
-
-    [nix-shell:~/cardano-sl]$ stack setup
-
-After that, in order to build Cardano SL with wallet capabilities, run the following script:
-
-    [nix-shell:~/cardano-sl]$ ./scripts/build/cardano-sl.sh
+## Nix build mode (recommended)
 
-It is suggested having at least 8GB of RAM and some swap space for the build process. As the project is fairly large and GHC parallelizes builds very effectively, memory and CPU consumption during the build process is high. Please make sure you have enough free disk space as well.
+First, prerequisite: install Nix (full instructions at https://nixos.org/nix/download.html):
 
-After the project is built - it can take quite a long time -  the built binaries can be launched using the `stack exec` command. Let's discuss important binaries briefly before proceeding to the next step.
+    curl https://nixos.org/nix/install | sh
 
-### Build Commands without Nix
+Two steps remain, then:
 
-We **strongly recommend** you to use [Nix package manager](https://nixos.org/nix/download.html) to build Cardano SL, but it is possible to build it without Nix as well.
+1.  To employ the signed IOHK binary cache:
 
-Clone Cardano SL repository and go to the root directory:
+        $ sudo mkdir -p /etc/nix
+        $ sudo vi /etc/nix/nix.conf       # ..or any other editor, if you prefer
 
-    $ git clone git@github.com:input-output-hk/cardano-sl.git
-    $ cd cardano-sl
+    ..and then add two following lines:
 
-Switch to the latest release branch, for example, `cardano-sl-1.0`:
+        $ binary-caches             = https://cache.nixos.org https://hydra.iohk.io
+        $ binary-caches-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
 
-    $ git checkout cardano-sl-1.0
+2.  Actually building Cardano SL (or, most likely, simply obtaining it from the IOHK's binary caches) can be performed by:
 
-And if it is the first project in Haskell on this machine, run `stack setup`:
+        $ nix-build -A cardano-sl-static --cores 0 --jobs 2 --no-build-output --out-link cardano-sl-1.0
 
-    $ stack setup
+    The build output directory will be symlinked as `cardano-sl-1.0` (as specified by the command).
```

Comment:
We are still missing a list of commonly needed executables here. For example: `cardano-node`, `cardano-explorer`, `cardano-node-simple` and mention that most of other tools are in `cardano-sl-tools`.